runtime/capsules/runtime/src/mctp/mctp_control.rs

```diff
@@ -0,0 +1,276 @@
+// Licensed under the Apache-2.0 license
+
+use bitfield::bitfield;
+use kernel::ErrorCode;
+use zerocopy::{FromBytes, Immutable, IntoBytes};
+
+pub const MCTP_CTRL_MSG_HEADER_LEN: usize = 3;
+
+const MCTP_NULL_EID: u8 = 0;
+const MCTP_BROADCAST_EID: u8 = 0xFF;
+
+bitfield! {
+    #[repr(C)]
+    #[derive(Clone, FromBytes, IntoBytes, Immutable)]
+    pub struct MCTPCtrlMsgHdr([u8]);
+    impl Debug;
+    u8;
+    pub ic, _: 0, 0;
+    pub msg_type, _: 7, 1;
+    pub rq, set_rq : 8, 8;
+    pub datagram, set_datagram: 9, 9;
+    rsvd, _: 10, 10;
+    pub instance_id, set_instance_id: 15, 11;
+    pub cmd, set_cmd: 23, 15;
+}
+
+impl MCTPCtrlMsgHdr<[u8; MCTP_CTRL_MSG_HEADER_LEN]> {
+    pub fn new() -> Self {
+        MCTPCtrlMsgHdr([0; MCTP_CTRL_MSG_HEADER_LEN])
+    }
+
+    pub fn prepare_header(&mut self, rq: u8, datagram: u8, instance_id: u8, cmd: u8) {
+        self.set_rq(rq);
+        self.set_datagram(datagram);
+        self.set_instance_id(instance_id);
+        self.set_cmd(cmd);
+    }
+}
+
+pub enum MCTPCtrlCmd {
+    SetEid,
+    GetEid,
+    GetMsgTypeSupport,
+    Unsupported,
+}
+
+impl From<u8> for MCTPCtrlCmd {
+    fn from(val: u8) -> MCTPCtrlCmd {
+        match val {
+            1 => MCTPCtrlCmd::SetEid,
```

Comment:
I always think "Eid Mubarak" when I see this :)

Reply:
Haha! I could use EID instead of Eid. Or maybe call it EndpointID, but that seems a bit too long.

Reply:
`Eid` is fine :) Just funny.